Sedah/Assets/Scripts/Character/Health.cs

```diff
@@ -151,18 +152,26 @@ public float TakeDamage(DamageInfo damageInfo)
     }
 
     // Start is called before the first frame update
-    public override void OnStartServer()
+    public void Start()
     {
-        base.OnStartClient();
-        this.character = this.GetComponent<CharacterObject>();
-        this.currentHealth = this.maxHealth;
-        this.lastHitTime = float.NegativeInfinity;
+
+    }
+
+    public void SetHealth()
```

Comment:
What is the idea behind doing this here instead of with the start method?

Reply:
The script is disabled when the player prefab is first spawned. I think that OnStartServer isn't called if you activate the script, so this is just a trigger to set the values after the script is activated.